pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -469,14 +481,23 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			expect: true,
 		},
 		{
-			description: "if the update strategy changes",
+			description: "if the update strategy's max unavailable parameter changes",
 			mutate: func(daemonset *appsv1.DaemonSet) {
-				daemonset.Spec.UpdateStrategy = appsv1.DaemonSetUpdateStrategy{
-					Type: appsv1.RollingUpdateDaemonSetStrategyType,
-					RollingUpdate: &appsv1.RollingUpdateDaemonSet{
-						MaxUnavailable: pointerTo(intstr.FromString("10%")),
-					},
-				}
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if the update strategy's max surge parameter changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxSurge = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if spec.minReadySeconds changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.MinReadySeconds = 9
```

Comment:
I'm confused, this doesn't change the default value?

Reply:
I'm confused because it is expected to change, but that is not a change.

Reply:
The old value doesn't specify `minReadySeconds`: https://github.com/openshift/cluster-dns-operator/blob/b0ca398b84de1287fe02d22bf9311f0053a02d1c/pkg/operator/controller/controller_dns_daemonset_test.go#L566-L621

Reply:
So, this is sort of a test of an upgrade to this feature?  Because a new installation would add the "assets" from assets/dns/daemonset.yaml, right?

Reply:
> The old value doesn't specify minReadySeconds

Ack. This isn't the first time this has tripped me up.

Reply:
> So, this is sort of a test of an upgrade to this feature? Because a new installation would add the "assets" from assets/dns/daemonset.yaml, right?

Yeah, that's one way to look at it.  The point of the test case is to verify that `TestDaemonsetConfigChanged` behaves correctly when the `minReadySeconds` field changes.